Sources/NIOSSL/SSLContext.swift

```diff
@@ -569,6 +567,40 @@ extension NIOSSLContext {
             parentSwiftContext.keyLogManager!.log(linePointer)
         }
     }
+    
+    /// Takes a path and determines if the file at this path is of c_rehash format .
+    internal static func _isRehashFormat(path: String) throws -> Bool {
+        // Check if the element’s name matches the c_rehash symlink name format.
+        // Note that `lastPathComponent` is not available here.
+        guard let lastPathComponent = path.split(separator: "/").last else { return false }
+        let filenameElements = lastPathComponent.split(separator: ".")
+        
+        guard filenameElements.count == 2,
+              let filenameElement = filenameElements.first,
+              let fileExtensionElement = filenameElements.last else {
+            return false
+        }
+        
+        // The links created are of the form HHHHHHHH.D, where each H is a hexadecimal character and D is a single decimal digit.
+        let filename = String(filenameElement)
+        let fileExtension = Int(fileExtensionElement)
+        
+        // Double check that the extension did not fail to cast to an integer.
+        // Make sure that the filename is an 8 character hex based file name.
+        guard fileExtension != nil,
+              filename.count == 8,
+              filename.utf8.allSatisfy( { Character(UnicodeScalar($0)).isHexDigit }) else { return false }
```

Comment:
Promoting things back up to `Character` isn't really getting us a meaningful performance win. We'd be better served by rewriting `isHexDigit` against `UTF8.CodeUnit`:

```swift
extension UTF8.CodeUnit {
    private static let asciiZero = UInt8(ascii: "0")
    private static let asciiNine = UInt8(ascii: "9")
    private static let asciiLowercaseA = UInt8(ascii: "a")
    private static let asciiLowercaseF = UInt8(ascii: "f")
    private static let asciiUppercaseA = UInt8(ascii: "A")
    private static let asciiUppercaseF = UInt8(ascii: "F")

    var isHexDigit: Bool {
        switch self {
        case (.asciiZero)...(.asciiNine),
             (.asciiLowercaseA)...(.asciiLowercaseF),
             (.asciiUppercaseA)...(.asciiUppercaseF):
            return true
        default:
            return false
        }
    }
}
```

Reply:
Wow, that is nice.  Added it to the bottom of `SSLContext` and refactored the usage of `Character`.